### PR TITLE
Add deeplink support

### PR DESCRIPTION
**Pull request by betrisey**
- Add a share button on the detailed view, it generates and share a link
- When the link is clicked, the app is opened and the item displayed

### CODE REVIEW

File: app/src/main/java/com/example/sharingang/MainActivity.kt [hunk 1]
```diff
@@ -42,10 +50,35 @@ class MainActivity : AppCompatActivity(R.layout.activity_main) {
...
 
+    /*
+    We don't have an option menu yet
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         val navController = getNavController()
         return item.onNavDestinationSelected(navController) || super.onOptionsItemSelected(item)
     }
-}
\ No newline at end of file
+    */
```
Comment (bafavre) [inline app/src/main/java/com/example/sharingang/MainActivity.kt]: Should we keep commented code?

File: app/src/main/java/com/example/sharingang/items/ItemsViewModel.kt [hunk 2]
```diff
@@ -139,7 +133,7 @@ class ItemsViewModel @Inject constructor(
             OBSERVABLES.SEARCH_RESULTS -> searchResults
         }
         observable.observe(LifeCycleOwner, {
-            it?.let { adapter.submitList(it) }
+            adapter.submitList(it)
```
Comment (bafavre) [inline app/src/main/java/com/example/sharingang/items/ItemsViewModel.kt:136]: What happens if "it" is null here? Everything is good or "it" cannot be null or ?

**Review by bafavre — APPROVED** [review]
Well done, I just had a few questions in the code :)

File: app/src/main/java/com/example/sharingang/items/ItemsViewModel.kt (hunk 2 above)
Comment (betrisey) [reply app/src/main/java/com/example/sharingang/items/ItemsViewModel.kt:136]: "it" is a List<Item> and not a List<Item>?
So Android Studio suggested to remove the check

File: app/src/main/java/com/example/sharingang/MainActivity.kt (hunk 1 above)
Comment (betrisey) [reply app/src/main/java/com/example/sharingang/MainActivity.kt]: I think I'll remove it, thanks

File: app/src/main/java/com/example/sharingang/items/ItemsViewModel.kt (hunk 2 above)
Comment (bafavre) [reply app/src/main/java/com/example/sharingang/items/ItemsViewModel.kt:136]: Oh yeah then makes sense, all good!

File: app/src/main/java/com/example/sharingang/MapFragment.kt [hunk 3]
```diff
@@ -138,11 +138,9 @@ class MapFragment : Fragment(), OnMapReadyCallback {
             map?.animateCamera(CameraUpdateFactory.newLatLng(marker.position))
             if (marker != lastLocationMarker) {
                 val temp: Item = marker.tag as Item
-                viewModel.onViewItem(temp)
                 this.findNavController().navigate(
-                    MapFragmentDirections.actionMapFragmentToDetailedItemFragment()
+                    MapFragmentDirections.actionMapFragmentToDetailedItemFragment(temp)
```
Comment (Superdoude1999) [inline app/src/main/java/com/example/sharingang/MapFragment.kt:142]: not important, but maybe find a more suitable name than temp

**Review by Superdoude1999 — APPROVED** [review]

**Pull request merged**